Subject: [PATCH 29/54] OMS7-N: ApplicationsState: add filter for Substratum icon overlays [1/2]

This commit allows the framework to handle the filtering of the icon overlays found for OMS.

diff --git a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
index 6fecd4044ca..d3a6e217735 100644
--- a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
+++ b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
@@ -1329,6 +1329,17 @@ public class ApplicationsState {
         }
     };
 
+    public static final AppFilter FILTER_SUBSTRATUM_ICONS = new AppFilter() {
+        public void init() {
+        }
+
+        @Override
+        public boolean filterApp(AppEntry entry) {
+            return !((entry.info.metaData != null) &&
+                    (entry.info.metaData.getString("Substratum_IconPack") != null));
+        }
+    };
+
     public static final AppFilter FILTER_WORK = new AppFilter() {
         private int mCurrentUser;
 
